Activate TracingContext earlier

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98461
* __->__ #100043

Ensure any calls into VariableTracker have a valid TC

Previously, Calls into VariableBuilder from symbolic locals construction were not done under an active TC.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

wconstab committed:
Activate TracingContext earlier

Ensure any calls into VariableTracker have a valid TC

[ghstack-poisoned]

wconstab commented on torch/_dynamo/symbolic_convert.py:
```diff
@@ -1909,70 +1909,73 @@ def __init__(
-                )(f_locals[k]),
+        # as soon as we create the tracing context we should keep it active, so any calls
+        # into dynamo apis can rely on finding it
+        with tracing(self.output.tracing_context):
```
@voznesenskym I think it might actually be a cleaner change if I could activate TracingContext outside the InstructionTranslator init call in convert_frame.py.

However, to do that i'd have to move the FakeMode and ShapeEnv constructors also out to convert_frame.py.  I am not actually sure if that's a bad change, lmk if you prefer that way or this PR.

anijain2305 replied:
Just trying to understand - Why this is better?

wconstab replied:
Updated pr desc to clarify, lmk if still not clear.

anijain2305 reviewed (approved):
Makes sense to me. Would also wait for @voznesenskym to make a comment.

wconstab committed:
Update on "Activate TracingContext earlier"


Ensure any calls into VariableTracker have a valid TC

Previously, Calls into VariableBuilder from symbolic locals construction were not done under an active TC.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire 

[ghstack-poisoned]
